src/openforms/submissions/models/submission.py

```diff
@@ -339,6 +339,20 @@ class Meta:
                 ),
                 name="registration_status_consistency_check",
             ),
+            models.CheckConstraint(
```

Comment:
Should we have a test for this constraint, or are we confident it will work as expected?

Reply:
we have tests that verify the registration backend key is properly persisted after logic evaluation, and we have plenty of tests that otherwise save the submission, so if it didn't work, there would've been a lot of test failures :P it works ;)